cpp/oneapi/dal/backend/primitives/select.cpp

```diff
@@ -0,0 +1,155 @@
+                iwspace[lrid + shift] = irow[shift];
+                dwspace[lrid + shift] = drow[shift];
+            }
+            idx.barrier(cl::sycl::access::fence_space::local_space);
+            // TODO: Optimize with async_work_group_copy
+        }
+
+        {
+            const idx_t k_step = k_width - k;
+            // Loop along vector size
+            for (idx_t shift = lxid; shift < vector_size; shift += k_step) {
+                // Copying data
+                for (idx_t i = 0; i < k_step; i += xrange) {
+                    iwspace[lrid + k + i] = shift + i;
+                    dwspace[lrid + k + i] =
+                        input_norms[shift + i] - 2 * input_cross[vector_size * gyid + shift + i];
+                }
+                idx.barrier(cl::sycl::access::fence_space::local_space);
+                // TODO: Another sort?
+                // Odd-Even Sorting
+                {
+                    // Temporary variables for swap
+                    Float tdst;
+                    idx_t tidx;
+                    // Sorting itself
+                    for (idx_t step = 0; step < k_width; ++step) {
+                        for (idx_t i = lxid + step % 2; i + 1 < k_width; i += xrange) {
+                            if (dwspace[lrid + i] > dwspace[lrid + i + 1]) {
+                                // TODO: Another swaping?
+                                {
+                                    tdst                  = dwspace[lrid + i];
+                                    dwspace[lrid + i]     = dwspace[lrid + i + 1];
+                                    dwspace[lrid + i + 1] = tdst;
+                                }
+                                {
+                                    tidx                  = iwspace[lrid + i];
+                                    iwspace[lrid + i]     = iwspace[lrid + i + 1];
+                                    iwspace[lrid + i + 1] = tidx;
+                                }
+                            }
+                        }
+                        idx.barrier(cl::sycl::access::fence_space::local_space);
+                    }
+                }
+            }
+        }
+
+        //Exit from algorithm
+        {
+            const idx_t* const irow = &(output_indices[gyid * k]);
+            const Float* const drow = &(ouput_distances[gyid * k]);
+            for (idx_t shift = lxid; shift < k; shift += xrange) {
+                irow[shift] = iwspace[lyid * k_width + shift];
+                drow[shift] = dwspace[lyid * k_width + shift];
+            }
+            // TODO: Optimize with async_work_group_copy
+        }
+    }
+
+private:
+    const Float* const input_cross;
+    const Float* const input_norms;
+    idx_t* const output_indices;
+    Float* const output_distances;
+    const idx_t k, k_width, vector_size;
+    dist_acc_t dwspace;
+    idxs_acc_t iwspace;
+};
+
+} // namespace impl
+
+#endif
+
+} // namespace oneapi::dal::backend::primitives
```

Comment:
The same comments as for reducer + add a blank line at the end

Reply:
Fixed in new PR